enumerables.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each
+    length.times do |v|
+      yield(self[v])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |v|
+      yield(self[v], v)
+    end
+  end
+
+  def my_select
+    arr = []
+    length.times do |v|
+      arr.push(self[v]) if yield(self[v])
+    end
+    arr
+  end
+
+  def my_all?
+    result = true
+    length.times do |v|
+      result = false if yield(self[v]) != true
+      break unless result == true
+    end
+    result
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true when none of the collection members are false or nil
```ruby
true_array = [1, true, 'hi', []] 
false_array = [1, false, 'hi', []] 
true_array.my_all? == true_array.all? #true
```
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
```ruby
array = Array.new(100){rand(0...9)}
array.my_all?(Integer) == array.all?(Integer)
```
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
```ruby
words = %w[dog door rod blade]
words.my_all?(/d/) == words.all?(/d/)
```
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
```ruby
array.my_all?(3) == array.all?(3) #true
```